// nest-shared/lib/config/application.config.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.configService = exports.ConfigService = void 0;
require("dotenv/config");
const common_1 = require("@nestjs/common");
const shared_1 = require("../shared");
let ConfigService = class ConfigService {
    constructor(env) {
        this.env = env;
    }
    getValue(key, throwOnMissing = true) {
        const value = this.env[key];
        const nodeEnv = process.env['NODE_ENV'];
        const safeEnv = nodeEnv == 'development' || nodeEnv === 'testing';
        const canBeThrown = safeEnv && !value && throwOnMissing;
        if (canBeThrown) {
            throw new Error(`ConfigService error - Missing env.${key}`);
        }
        return value;
    }
    getBoolean(key) {
        const value = this.getValue(key);
        return value === 'true';
    }
    ensureValues(keys) {
        keys.forEach((k) => this.getValue(k, true));
        return this;
    }
    getPort() {
        const value = this.getValue('PORT', false);
        return value || shared_1.NODE_PORT;
    }
    isProduction() {
        const mode = this.getEnvironment();
        return mode === 'production';
    }
    isDevelopment() {
        const mode = this.getEnvironment();
        return mode === 'development';
    }
    isTest() {
        const mode = this.getEnvironment();
        return mode === 'testing';
    }
    getEnvironment() {
        const value = this.getValue('NODE_ENV', true);
        return String(value);
    }
};
ConfigService = __decorate([
    (0, common_1.Injectable)()
], ConfigService);
exports.ConfigService = ConfigService;
exports.configService = new ConfigService(process.env);
exports.default = exports.configService;
