// nest-shared/lib/modules/file/services/file.service.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileService = void 0;
const node_crypto_1 = __importDefault(require("node:crypto"));
const node_url_1 = require("node:url");
const platform_express_1 = require("@nestjs/platform-express");
const AWS = __importStar(require("aws-sdk"));
const multer_s3_1 = __importDefault(require("multer-s3"));
const unique_slug_1 = __importDefault(require("unique-slug"));
class FileService {
    constructor(args) {
        this.args = args;
        this.s3 = new AWS.S3();
        this.genFileName = (req, { originalname }, callback) => {
            void req;
            const [, extension] = originalname.split('.');
            callback(null, `${node_crypto_1.default.randomUUID()}.${extension}`);
        };
        AWS.config.update({
            accessKeyId: this.args.AWS_ACCESS_KEY_ID,
            secretAccessKey: this.args.AWS_SECRET_ACCESS_KEY,
        });
    }
    filesInterceptor(args) {
        var _a;
        const bucket = args.bucket;
        const metadata = (req, file, cb) => {
            void req;
            cb(null, { fieldName: file.fieldname });
        };
        const key = (req, file, cb) => {
            this.genFileName(req, file, (err, fileName) => {
                void err;
                cb(null, (0, unique_slug_1.default)(fileName));
            });
        };
        const contentType = (req, file, cb) => {
            void req;
            cb(null, file.mimetype);
        };
        return (0, platform_express_1.FilesInterceptor)(args.field_name, Number((_a = args.options) === null || _a === void 0 ? void 0 : _a.maxCount), {
            storage: (0, multer_s3_1.default)({
                s3: this.s3,
                bucket: String(bucket),
                acl: args === null || args === void 0 ? void 0 : args.acl,
                metadata: metadata,
                key: key,
                contentType: contentType,
            }),
        });
    }
    getS3FileURL(req) {
        var _a;
        const files = req.files;
        if (!files || !files.length) {
            return null;
        }
        return new node_url_1.URL((_a = files[0]) === null || _a === void 0 ? void 0 : _a.location);
    }
    getS3FileKey(req) {
        var _a;
        const fileURL = this.getS3FileURL(req);
        if (!fileURL || !fileURL.pathname) {
            return null;
        }
        const parsedPath = (_a = fileURL === null || fileURL === void 0 ? void 0 : fileURL.pathname) === null || _a === void 0 ? void 0 : _a.replace(/^\/+/, '');
        return String(parsedPath);
    }
    getSignedUrl(key, expires, bucket) {
        const signedURl = this.handleSignedUrl(String(bucket), key, expires || 60 * 5);
        return signedURl;
    }
    handleSignedUrl(bucket, key, expires) {
        const url = this.s3.getSignedUrl('getObject', {
            Bucket: bucket,
            Key: key,
            Expires: expires,
        });
        return new node_url_1.URL(url);
    }
    parseFileURL(avatar, bucket, expires) {
        const signedUrl = this.getSignedUrl(avatar, expires || 3600, bucket);
        return String(signedUrl.href);
    }
}
exports.FileService = FileService;
exports.default = FileService;
