// randts/lib/shared/config/Configuration.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Configuration = void 0;
const NumberLength_1 = __importDefault(require("../../core/NumberLength"));
class Configuration {
    constructor() {
        this._timestampBased = false;
        this._minLength = new NumberLength_1.default(0);
        this._maxLength = new NumberLength_1.default(0);
        this.reset();
    }
    reset() {
        this._minLength = new NumberLength_1.default(0);
        this._maxLength = new NumberLength_1.default(0);
        this._timestampBased = false;
        return this;
    }
    setLength(length) {
        this._validateLength(length);
        const Length = new NumberLength_1.default(length);
        this._minLength = new NumberLength_1.default(Length.getValue());
        this._maxLength = new NumberLength_1.default(Length.getValue());
        return this;
    }
    setMinLength(length) {
        this._validateLength(length);
        this._minLength = new NumberLength_1.default(length);
        return this;
    }
    getMinLength() {
        if (!this._minLength) {
            this._minLength = new NumberLength_1.default();
        }
        if (this._maxLength && this._minIsGreaterThanMax()) {
            this._minLength = new NumberLength_1.default(this._maxLength.getValue());
        }
        return this._minLength;
    }
    setMaxLength(length) {
        this._validateLength(length);
        this._maxLength = new NumberLength_1.default(length);
        return this;
    }
    getMaxLength() {
        if (!this._maxLength) {
            this._maxLength = new NumberLength_1.default();
        }
        if (this._minLength && this._minIsGreaterThanMax()) {
            this._maxLength = new NumberLength_1.default(this._minLength.getValue());
        }
        return this._maxLength;
    }
    timestampBased() {
        this._timestampBased = true;
        return this;
    }
    isTimestampBased() {
        return Boolean(this._timestampBased);
    }
    _validateLength(length) {
        if (length && typeof length !== 'number') {
            throw new Error('The length param must be a number');
        }
        return true;
    }
    _minIsGreaterThanMax() {
        return this._minLength.getValue() > this._maxLength.getValue();
    }
}
exports.Configuration = Configuration;
exports.default = Configuration;
