// nest-shared/lib/shared/helpers/crypto/Base64.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.decode = exports.encode = exports.validateUUID = exports.decodeToBuffer = exports.base64Gen = exports.randomUUID = exports.base64 = exports.Base64 = void 0;
const node_crypto_1 = __importDefault(require("node:crypto"));
const base64_1 = require("@localtools/base64");
const constants_1 = require("../../constants");
class Base64 {
    decodeToBuffer(str, encoding = 'utf8') {
        return Buffer.from(str.toString(), encoding);
    }
    validateUUID(uuid) {
        return constants_1.VALID_UUID_REGEX.test(uuid);
    }
}
exports.Base64 = Base64;
exports.base64 = new Base64();
exports.randomUUID = node_crypto_1.default.randomUUID();
exports.base64Gen = base64_1.Base64.encode({
    text: exports.randomUUID,
});
exports.decodeToBuffer = exports.base64.decodeToBuffer;
exports.validateUUID = exports.base64.validateUUID;
exports.encode = base64_1.Base64.encode;
exports.decode = base64_1.Base64.decode;
exports.default = exports.base64;
