// nestjs-firebase-admin/dist/modules/admin/admin.service.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminService = void 0;
const common_1 = require("@nestjs/common");
const rxjs_1 = require("rxjs");
const app_1 = require("firebase-admin/app");
const firebase_admin_1 = __importDefault(require("firebase-admin"));
const admin_constants_1 = require("./admin.constants");
let AdminService = exports.AdminService = class AdminService {
    constructor(options) {
        this.options = options;
        firebase_admin_1.default.initializeApp(Object.assign(Object.assign({}, this.options), { credential: firebase_admin_1.default.credential.cert(this.options.credential) }));
    }
    applicationDefault(httpAgent) {
        return (0, app_1.applicationDefault)(httpAgent);
    }
    deleteApp(app) {
        return (0, app_1.deleteApp)(app);
    }
    get getApps() {
        return (0, app_1.getApps)();
    }
    get getApp() {
        return (0, app_1.getApp)();
    }
    admin() {
        return firebase_admin_1.default;
    }
    initializeAppObservable() {
        return new rxjs_1.Observable(subscriber => {
            subscriber.next(this.appRef);
            subscriber.complete();
            return () => {
                if (this.appRef.name) {
                    return this.appRef;
                }
                return this.getApp;
            };
        });
    }
    get appRef() {
        return this.initializeApp();
    }
    initializeApp() {
        return this.admin().initializeApp(Object.assign(Object.assign({}, this.options), { credential: firebase_admin_1.default.credential.cert(this.options.credential) }));
    }
};
exports.AdminService = AdminService = __decorate([
    (0, common_1.Injectable)(),
    __param(0, (0, common_1.Inject)(admin_constants_1.FIREBASE_ADMIN_INSTANCE_TOKEN)),
    __metadata("design:paramtypes", [Object])
], AdminService);
