// randts/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RandNum = exports.RandomNumber = exports.NumberLength = exports.Configuration = void 0;
const Configuration_1 = __importDefault(require("./shared/config/Configuration"));
const NumberLength_1 = __importDefault(require("./core/NumberLength"));
const AbstractRandomNumber_1 = __importDefault(require("./core/AbstractRandomNumber"));
const RandomNumber_1 = __importDefault(require("./core/RandomNumber"));
const RandomNumberTimestampBased_1 = __importDefault(require("./core/RandomNumberTimestampBased"));
var Configuration_2 = require("./shared/config/Configuration");
Object.defineProperty(exports, "Configuration", { enumerable: true, get: function () { return Configuration_2.Configuration; } });
var NumberLength_2 = require("./core/NumberLength");
Object.defineProperty(exports, "NumberLength", { enumerable: true, get: function () { return NumberLength_2.NumberLength; } });
var RandomNumber_2 = require("./core/RandomNumber");
Object.defineProperty(exports, "RandomNumber", { enumerable: true, get: function () { return RandomNumber_2.RandomNumber; } });
class RandNum {
    constructor(NumberConfiguration = undefined) {
        this._number = this._generate(NumberConfiguration);
    }
    getNumber() {
        return this._number;
    }
    _generate(NumberConfiguration) {
        if (!(NumberConfiguration instanceof Configuration_1.default)) {
            NumberConfiguration = this._generateConfiguration();
        }
        return NumberConfiguration.isTimestampBased()
            ? new RandomNumberTimestampBased_1.default(NumberConfiguration)
            : new RandomNumber_1.default(NumberConfiguration);
    }
    _generateConfiguration() {
        const NumberConfiguration = new Configuration_1.default();
        return NumberConfiguration.setLength(new NumberLength_1.default().getValue());
    }
}
exports.RandNum = RandNum;
exports.default = {
    Configuration: Configuration_1.default,
    NumberLength: NumberLength_1.default,
    AbstractRandomNumber: AbstractRandomNumber_1.default,
    Generator: RandNum,
};
