// date-handle/lib/core/date-handle.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DateHandle = void 0;
const moment_1 = __importDefault(require("moment"));
class DateHandle {
    constructor() {
        this.day = (0, moment_1.default)().weekday();
    }
    dayOfWeek(thisDay) {
        return this.day === thisDay;
    }
    isSaturday() {
        return this.dayOfWeek(6);
    }
    isSunday() {
        return this.dayOfWeek(0);
    }
    isMonday() {
        return this.dayOfWeek(1);
    }
    add(date, options) {
        const { days = 0, months = 0, years = 0 } = options;
        const newDate = new Date(date);
        newDate.setDate(newDate.getDate() + days);
        newDate.setMonth(newDate.getMonth() + months);
        newDate.setFullYear(newDate.getFullYear() + years);
        return newDate;
    }
    difference(start_date, end_date) {
        const diff = new Date(end_date).getTime() - new Date(start_date).getTime();
        return new Date(diff);
    }
    static future(amount, unit) {
        const future = (0, moment_1.default)().add(amount, unit).format('YYYY-MM-DD');
        return (0, moment_1.default)(future);
    }
    static futureFully(amount, unit) {
        const future = (0, moment_1.default)().add(amount, unit).format('YYYY-MM-DD HH:mm:ss');
        return (0, moment_1.default)(future);
    }
    static past(amount, unit) {
        const past = (0, moment_1.default)().add(amount, unit).format('YYYY-MM-DD');
        return (0, moment_1.default)(past);
    }
    static pastFully(amount, unit) {
        const past = (0, moment_1.default)().add(amount, unit).format('YYYY-MM-DD HH:mm:ss');
        return (0, moment_1.default)(past);
    }
}
exports.DateHandle = DateHandle;
DateHandle.date = {
    y: (0, moment_1.default)().format('YYYY'),
    m: (0, moment_1.default)().format('MM'),
    d: (0, moment_1.default)().format('DD'),
    now: (0, moment_1.default)().format('YYYY-MM-DD'),
    nowFully: (0, moment_1.default)().format('YYYY-MM-DD HH:mm:ss'),
    h: (0, moment_1.default)().format('HH:mm:ss'),
};
DateHandle.yesterday = () => (0, moment_1.default)().subtract(1, 'days').format('YYYY-MM-DD');
